Extend macOS and iOS sandboxes to enable CoreAudio logging in GPUProcess

youennf opened the pull request:
#### 73b9c9c738c159611f3dde1c9fe26f282e0fff40
<pre>
Extend macOS and iOS sandboxes to enable CoreAudio logging in GPUProcess
<a href="https://bugs.webkit.org/show_bug.cgi?id=242937">https://bugs.webkit.org/show_bug.cgi?id=242937</a>
rdar://76778043

Reviewed by Per Arne Vollan.

* Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
* Source/WebKit/Resources/SandboxProfiles/iOS/com.apple.WebKit.GPU.sb.in:

Canonical link: <a href="https://commits.webkit.org/252785@main">https://commits.webkit.org/252785@main</a>
</pre>

pvollan commented on Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
```diff
@@ -845,7 +845,14 @@
+    (allow user-preference-read (preference-domain "com.apple.coreaudio"))
+    (allow file-read* file-write*
+        (subpath "/Library/Preferences/Audio")
+        (subpath "/tmp/AudioCapture/VP"))
```
Would it be sufficient to allow this only on internal builds?

(with-filter (system-attribute apple-internal)
...

youennf replied:
Good idea, I'll update accordingly

pvollan reviewed (approved):
R=me.

youennf committed:
Extend macOS and iOS sandboxes to enable CoreAudio logging in GPUProcess
https://bugs.webkit.org/show_bug.cgi?id=242937
rdar://76778043

Reviewed by Per Arne Vollan.

* Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
* Source/WebKit/Resources/SandboxProfiles/iOS/com.apple.WebKit.GPU.sb.in:

Canonical link: https://commits.webkit.org/252785@main
